torchtune/utils/_generation.py

```diff
@@ -4,141 +4,115 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Callable, Optional, Set
+from typing import Callable, List, Optional
 
 import torch
-
 from torchtune.modules import TransformerDecoder
 
 
 def multinomial_sample_one(probs):
+    """Samples from a multinomial distribution."""
     q = torch.empty_like(probs).exponential_(1)
     return torch.argmax(probs / q, dim=-1, keepdim=True).to(dtype=torch.int)
 
 
-def sample(
-    logits: torch.Tensor, temperature: float = 1.0, top_k: Optional[int] = None
-) -> torch.Tensor:
-    # scale the logits based on temperature
+def sample(logits, temperature=1.0, top_k=None):
+    """Samples from a probability distribution."""
     logits = logits / max(temperature, 1e-5)
-
-    # keep only the top_k logits if this is specified
     if top_k is not None:
         v, _ = torch.topk(logits, min(top_k, logits.size(-1)))
-        # select the very last value from the top_k above as the pivot
         pivot = v.select(-1, -1).unsqueeze(-1)
-        # set everything smaller than pivot value to inf since these
-        # should be pruned
         logits = torch.where(logits < pivot, -float("Inf"), logits)
-
-    # compute the probabilities
     probs = torch.nn.functional.softmax(logits, dim=-1)
+    return multinomial_sample_one(probs)
 
-    # sample the next token
-    token = multinomial_sample_one(probs)
-    return token
-
-
-def generate_next_token(
-    model: TransformerDecoder,
-    input_pos: torch.Tensor,
-    x: torch.Tensor,
-    temperature: float = 1.0,
-    top_k: Optional[int] = None,
-) -> torch.Tensor:
-    # x: [1, s]
-    # input_pos: [s]
-    logits = model(x, input_pos)
-
-    # logits: [1, s, v] where v is vocab_size
-    # for sampling we extract the logits for the
-    # last token and convert to shape: [v]
```

Comment:
Please don't delete the shapes. It's really hard to understand this without this information. Maybe these need to be updated, but I'd like to keep these